myhpom/views/choose_network.py

```diff
@@ -16,29 +16,28 @@ def choose_network(request):
     """
     user_details = request.user.userdetails
     state = user_details.state
+    form = ChooseNetworkForm()
 
-    if request.method == "POST":
+    if request.POST:
         form = ChooseNetworkForm(request.POST)
-        custom_provider = (form.data.get("custom_provider") or '').strip()
-        health_network_id = form.data.get("healthcare_network")
-        health_network = None
-        if custom_provider != '':
-            # see if it actually already exists
-            networks = HealthNetwork.objects.filter(state=state, name=custom_provider.strip())
-            if len(networks) > 0:
-                network = networks[0]
-            else:
-                user_details.custom_provider = custom_provider
-                user_details.health_network = None
+        if form.is_valid():
```

Comment:
Karen brought up before that this particular form isn't using the standard `save()` mechanism for a model-based form, which would consolidate validation/clean/save logic. This should be a `ModelForm` for UserDetails, with a custom `save()` implementation (moving the logic under `if form.is_valid()` into `ChooseNetworkForm.save()`.